### ISSUE TITLE
Improve error stack

### ISSUE THREAD

**Issue by ISSOtm** [rednex/rgbds#392]
The current error stack isn't user-friendly at all. Here's an example:
```
x: macro
X = 0
	REPT \
	10
X = X + 1
		IF X == 2
			WARN "Oh noes"
		ENDC
	ENDR
endm

y: macro
	x
endm

 y
```

Current RGBASM prints this:
```
warning: test.asm(16) -> y(1) -> x(7) -> x(6):
    Oh noes
```
This means to look at the 1st line of `y` (which is where `x` is called), then the 7th line of `x` (which is, uhhhhh) and then the 1st line of `x` (what??)

After drafting a patch, here's what was agreed (so far) should be printed:
```
warning: test.asm(16) -> test.asm::y(13) -> test.asm::x(3) -> test.asm::x::REPT~2(7):
    Oh noes
```

First, if anyone has complaints or suggestions about this, please do so.

Second, should this be the default, or only occur when a new flag is set? The question is whether the old behavior would be desirable; what I'm afraid of is that conditionally enabling the behavior could complexify the patch to an extent where I'm not comfortable with anymore.

**Comment by mid-kid** [comment]
This looks good and there's no real reason to add a new flag for this when it wouldn't break any source files, nor any workflow since it doesn't omit any information.

**Comment by ISSOtm** [comment]
It could break workflows based on regex-parsing the error stack output. I'm wondering if those exist, and if yes, if the change is lightweight enough.

**Comment by mid-kid** [comment]
When you parse any kind of user-facing output you must assume it might break eventually. The only place where I can see this being a slight annoyance is people using Vim's `errorformat` (does anyone?). I really don't see the purpose of maintaining compatibility for this, since codebases that don't trigger any errors will continue functioning without a hitch, and people working with it can adapt rather easily.

Just remember to update the files in the `test/` directory when you do this.

**Comment by ISSOtm** [comment]
Ok then.

> Just remember to update the files in the test/ directory when you do this.

You have no idea. x)

**Comment by ISSOtm** [comment]
Submitted my draft as #395.

**Comment by dbrotz** [comment]
I believe this can be closed now that #395 has been merged.

**Comment by ISSOtm** [comment]
Indeed!
